app/models/fact_value.rb

```diff
@@ -81,13 +81,19 @@ def self.count_each(fact, options = {})
     output
   end
 
-  def self.build_facts_hash(facts)
-    hosts = Host.where(:id => facts.group_by(&:host_id).keys).all
-
+  def self.build_facts_hash(facts, host_id = nil)
     hash = {}
+    if host_id
+      hosts = Host.where(:id => host_id).empty? ? Host.where(:name => host_id) : Host.where(:id => host_id)
```

Comment:
@tbrisker, can you think of a better way for selecting a host either by its name or id?

Reply:
There are two options I think of - ```Host::Base.friendly.find(host_id)``` that relies on friendly_id gem, or ```Host::Base.where(id: host_id).or(Host::Base.where(name: host_id))``` that uses general activerecord methods.